cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -588,20 +614,85 @@ inline std::array<int64_t, 3> GetIsoCalendar(int64_t arg, Localizer&& localizer)
           static_cast<int64_t>(weekday(ymd).iso_encoding())};
 }
 
+template <typename Duration, typename InType>
+struct ISOCalendarWrapper {
+  static Status get(const Scalar& in, std::array<int64_t, 3>& iso_calendar) {
```

Comment:
I think this factorization is good, however, just a nit: methods should use UpperCamelCase unless they're const getters so this should be called Get.

Reply:
Done, thanks @lidavidm @rok !